### PR TITLE
Expose HttpZeroRttChecker, to be able to test ZeroRtt being rejected

### PR DESCRIPTION
**Pull request by ddragana**

### CODE REVIEW

**Comment by ddragana** [comment]
This is not the best API for a Http3Server, but I avoided to make something more complex just  to enable testing of ZeroRtt being rejected.

File: neqo-server/src/main.rs [hunk 1]
```diff
@@ -466,6 +466,11 @@ impl ServersRunner {
                     max_table_size_decoder: args.max_table_size_decoder,
                     max_blocked_streams: args.max_blocked_streams,
                 },
+                Some(HttpZeroRttChecker::new(QpackSettings {
+                    max_table_size_encoder: args.max_table_size_encoder,
+                    max_table_size_decoder: args.max_table_size_decoder,
+                    max_blocked_streams: args.max_blocked_streams,
+                })),
```
Comment (martinthomson) [inline neqo-server/src/main.rs:473]: How is this different than `None`?

**Review by martinthomson — COMMENTED** [review]
Did you plan to add a test that used this new option?  I don't see that code.

File: neqo-server/src/main.rs (hunk 1 above)
Comment (ddragana) [reply neqo-server/src/main.rs:473]: It is not. I used that for testing and wanted to remove it, but forgot.

**Comment by ddragana** [comment]
> Did you plan to add a test that used this new option? I don't see that code.

I need this option for Necko testing, but I will add tests here as well.

File: neqo-http3/src/server.rs [hunk 2]
```diff
@@ -54,14 +54,19 @@ impl Http3Server {
...
+        zero_rtt_checker: Option<Box<dyn ZeroRttChecker>>,
     ) -> Res<Self> {
         Ok(Self {
             server: Server::new(
                 now,
                 certs,
                 protocols,
                 anti_replay,
-                Box::new(HttpZeroRttChecker::new(qpack_settings)),
+                if let Some(c) = zero_rtt_checker {
```
Comment (martinthomson) [inline neqo-http3/src/server.rs:65]: This should use unwrap_or_else

File: neqo-http3/src/server.rs [hunk 3]
```diff
@@ -446,8 +457,8 @@ mod tests {
     }
 
     // Connect transport, send and receive settings.
-    fn connect() -> (Http3Server, PeerConnection) {
-        let (mut hconn, mut neqo_trans_conn) = connect_and_receive_settings();
+    fn create_client_and_connect_with_server(server: &mut Http3Server) -> PeerConnection {
```
Comment (martinthomson) [inline neqo-http3/src/server.rs:460]: A shorter name might be easier.

```suggestion
    fn connect_to(server: &mut Http3Server) -> PeerConnection {
```

**Review by martinthomson — COMMENTED** [review]
Fix and land.  Thanks for updating the test.  Good to see that doing that had the right effect.

**Pull request merged**